python/taichi/lang/ast/ast_transformer.py

```diff
@@ -1118,7 +1118,8 @@ def build_While(ctx, node):
                 "'else' clause for 'while' not supported in Taichi kernels")
 
         with ctx.loop_scope_guard():
-            ctx.ast_builder.begin_frontend_while(expr.Expr(1).ptr)
+            ctx.ast_builder.begin_frontend_while(
+                ti_ops.cast(1, primitive_types.i32).ptr)
```

Comment:
```suggestion
                expr.Expr(1, dtype=primitive_types.i32).ptr)
```
The cast can be avoided here actually.